### PR TITLE
[flutter_tools] write sksl on exit

### LINKED ISSUE #54484: Automatically write SkSLs with --cache-sksl flag
Instead of asking developer to manually press "M" during a `flutter run`, sometimes it's more ergonomic to automatically write SkSLs by the end of `flutter run` or `flutter drive` when `--cache-sksl` is provided.

The new workflow also makes it less likely to receive an empty set of SkSLs as `--cache-sksl` is required.

The old workflow would still be useful if developers aren't sure how to trigger a shader compilation so they can do something and press "M" iteratively.

Finally, the new workflow makes it easier for CI tests since they don't have to simulate a "M" key press.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Allow dumping sksl files on driver exit when a file path is provided to drive's --write-sksl-on-exit

Fixes https://github.com/flutter/flutter/issues/54484

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 1]
```diff
@@ -111,7 +114,11 @@ class DriveCommand extends RunCommandBase {
           'Works only if \'browser-name\' is set to \'android-chrome\'')
       ..addOption('chrome-binary',
         help: 'Location of Chrome binary. '
-          'Works only if \'browser-name\' is set to \'chrome\'');
+          'Works only if \'browser-name\' is set to \'chrome\'')
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/commands/drive.dart:117]: @liyuqian would it be easier if the command required you to pass in a destination file?

like --write-sksl-on-exit=foo.sksl.json ?

Comment (liyuqian) [reply packages/flutter_tools/lib/src/commands/drive.dart:117]: That sounds good to me. It also makes my test easier as I don't have to worry about whether the tool will create sksl_02.json or sksl_01.json. With `--write-sksl-on-exit=foo.sksl.json `, I'd expect the file to be overwritten, right?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/drive.dart:117]: Yup!

**Review by liyuqian — APPROVED** [review]
LGTM! I'll later test this with https://github.com/flutter/flutter/pull/56638 so we'll have some test coverages soon.

**Comment by jonahwilliams** [comment]
SG, I added a trivial unit test - your benchmark could be the integration test

**Pull request merged**